lib/webdriveragent.js

```diff
@@ -362,36 +371,32 @@ class WebDriverAgent {
 
     let agentUrl;
     let lineCount = 0;
-    let reachedEnd = !this.realDevice; // simulator does not need to wait, since we are tailing
     let showWaitingMessage = true; // turn off logging once we have hit the end
 
     let startDetector = (stdout) => {
-      // on a real device there may already be system logs that need to be
-      // passed before we get to the real startup logs, otherwise
-      // we expect two lines, one after another
-      //     Jul 20 13:03:57 iamPhone XCTRunner[296] <Warning>: Built at Jul 20 2016 13:03:50
-      //     Jul 20 13:03:57 iamPhone XCTRunner[296] <Warning>: ServerURLHere->http://10.35.4.122:8100<-ServerURLHere
-      if (!reachedEnd) {
-        let dateMatch = REAL_DEVICE_BUILD_LOG_STARTTIME_REGEX.exec(stdout);
-        if (dateMatch) {
-          let buildTime = new Date(dateMatch[1]);
-          if (buildTime.isAfter(startTime)) {
-            reachedEnd = true;
+      let match = AGENT_STARTED_REGEX.exec(stdout);
+      if (match) {
+        if (this.realDevice) {
+          // on a real device there may already be system logs that need to be
+          // passed before we get to the real startup logs, otherwise
+          // it will be "done" before booting
+          // Expect a line like:
+          //     Dec  7 10:55:32 iamPhone XCTRunner(WebDriverAgentLib)[386] <Notice>: ServerURLHere->http://localhost:8100<-ServerURLHere
+          // need to get the date and add the year from the start date
+          let dateString = `${stdout.substr(0, 6)} ${startTime.getFullYear()} ${stdout.substr(7, 8)}`;
+          let buildTime = new Date(dateString);
+          if (!buildTime.isAfter(startTime)) {
+            return false;
           }
         }
-      }
 
-      if (reachedEnd) {
-        let match = AGENT_STARTED_REGEX.exec(stdout);
-        if (match) {
-          agentUrl = match[1];
-          log.info(`Detected that WebDriverAgent is running at url '${agentUrl}'`);
-          if (!agentUrl) {
-            log.errorAndThrow(new Error('No url detected from WebDriverAgent'));
-          }
-          showWaitingMessage = false;
-          return true;
+        agentUrl = match[1];
+        log.info(`Detected that WebDriverAgent is running at url '${agentUrl}'`);
+        if (!agentUrl) {
```

Comment:
True. Though in practice I have not seen `agentUrl` ever come up empty, nor can I imagine how it would happen, given how WDA is generating the line.

But the check will make it seem more sensible.

Updated.